packages/kuma-gui/src/test-support/mocks/src/meshes/_/dataplanes/_/xds.ts

```diff
@@ -1,521 +1,717 @@
 import type { EndpointDependencies, MockResponder } from '@/test-support'
-export default (_deps: EndpointDependencies): MockResponder => (_req) => {
+export default ({ env, fake }: EndpointDependencies): MockResponder => (req) => {
+  const query = req.url.searchParams
+  const { name } = req.params
+  const eds = query.get('include_eds') === 'true'
+
+  // use a seed based on the name to keep ports and ip address the same across
+  // _overview, stats and rules
+  fake.kuma.seed(name as string)
```

Comment:
💡 👍

Reply:
👍 I should spend some time documenting this a little more. This is how we do state in a stateless mocking framework. The state is in the URL.